Slim/Control/Queries.pm

```diff
@@ -724,6 +728,12 @@ sub albumsQuery {
 		# Bind selected columns in order
 		my $i = 1;
 		for my $col ( @cols ) {
+		# Adjust column names that are sub-queries to be stored using the AS value
+			if ( $col =~ /SELECT/ ) {
+				my ($newcol) = $col =~ /AS (\w+)/;
+				$c->{$newcol} = 1;
+				$col = $newcol;
+			}
```

Comment:
I found this code in `_getTagDataForTracks`, designed to allow a sensible name for subselects,  but it didn't work first off, because of the fix for bug 15997 (see new line 651 above) meant we ended up with 2 "AS" clauses.

I've also corrected in `_getTagDataForTracks`, even though there are no subselects at present in there (presumably there used to be, for that code to exist). But PLEASE check carefully and test, I don't want to break anything!

Reply:
Ohh... that looks... fragile. Couldn't you store the desired `$newcol` value in `$c->{'SELECT ....'} = 'group_count';`? That way you'd use that value if it was anything other than `1`.

```perl
my $newcol = $c->{$col} if $c->{$col}) ne '1';
...
```

(or similar - to be tested)

Reply:
Hmm... I think I'd prefer something in between your former suggestion and mine. Using an array as a marker to treat it some different way looks wrong to me... 

You initially suggested to check for `AS` and something in the key and parse the column name out of this (which I didn't like). So I suggested setting the value correspondingly - which you found out was used differently before. Why not check the key as you suggested, but then use the value (as I suggested) instead of parsing the key? A key which has space and `AS` certainly wouldn't be a valid column name. So it should be save to do this. What do you think?